lib/util/timeout.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/**
+ * A {@link Timeout} represents a resettable and clearable timeout.
+ */
+class Timeout {
+  /**
+   * Construct a {@link Timeout}.
+   * @param {function} fn - Function to call
+   * @param {number} delay - Delay in milliseconds
+   * @param {Array<*>} args - Optional arguments to be passed to the function
+   */
+  constructor(fn, delay, ...args) {
```

Comment:
I recommend nixing the `args` functionality—It can be handled by passing an appropriate `fn`, which then simplifies Timeout.